pkg/sql/alter_primary_key.go

```diff
@@ -467,19 +452,6 @@ func (p *planner) AlterPrimaryKey(
 			newIndex.Partitioning = descpb.PartitioningDescriptor{}
 		}
 
-		newIndex.Name = tabledesc.GenerateUniqueConstraintName(basename, nameExists)
-		if err := addIndexMutationWithSpecificPrimaryKey(ctx, tableDesc, &newIndex, newPrimaryIndexDesc); err != nil {
-			return err
-		}
-		// If the index that we are rewriting is interleaved, we need to setup the rewritten
-		// index to be interleaved as well. Since we cloned the index, the interleave descriptor
-		// on the new index is already set up. So, we just need to add the backreference from the
-		// parent to this new index.
-		if len(newIndex.Interleave.Ancestors) != 0 {
-			if err := p.finalizeInterleave(ctx, tableDesc, &newIndex); err != nil {
-				return err
-			}
-		}
```

Comment:
The reason this needs to be pushed down is that `CreatePartitioning` can mutate `newIndex`, especially `ColumnIDs`, which in turn determines what makes it into `ExtraColumnIDs`. The latter is computed by `addIndexMutationWithSpecificPrimaryKey`.